### PR TITLE
Add GitHub Actions workflow

### LINKED ISSUE #737: Migrate from Travis to GitHub Actions
See https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works for context, Travis is no longer a feasible CI provider for open-source projects, so the CI setup needs to be moved over to GH actions.

### PR DESCRIPTION
**Pull request by mglaman**
Fixes #737 

I didn't delete `.travis.yml` in this PR. I figured it could linger or be a follow-up post merge.

### CODE REVIEW

**Comment by mglaman** [comment]
:/ hm, so `test_old/run-php-src.sh 8.0.0` keeps erroring with

```
Fatal error: Uncaught Error: Undefined constant 'T_DOUBLE_COLON' in /home/runner/work/PHP-Parser/PHP-Parser/lib/PhpParser/Lexer.php:476
Stack trace:
#0 /home/runner/work/PHP-Parser/PHP-Parser/lib/PhpParser/Lexer.php(40): PhpParser\Lexer->createTokenMap()
#1 /home/runner/work/PHP-Parser/PHP-Parser/lib/PhpParser/Lexer/Emulative.php(44): PhpParser\Lexer->__construct(Array)
#2 /home/runner/work/PHP-Parser/PHP-Parser/test_old/run.php(136): PhpParser\Lexer\Emulative->__construct(Array)
#3 {main}
  thrown in /home/runner/work/PHP-Parser/PHP-Parser/lib/PhpParser/Lexer.php on line 476
```

https://github.com/mglaman/PHP-Parser/actions/runs/407264322

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# https://help.github.com/en/categories/automating-your-workflow-with-github-actions
+name: Main
+on:
+  push:
+  pull_request:
+    branches:
+          - "master"
```
Comment (mglaman) [inline .github/workflows/main.yml:7]: This way it runs on all branch bushes, but only PRs against `master`.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      - name: "Install dependencies"
+        run: |
+          composer require php-coveralls/php-coveralls:^2.2 --dev --no-update
+          composer update --no-progress --prefer-dist
+      - name: "Tests"
+        run: "php vendor/bin/phpunit --coverage-clover build/logs/clover.xml"
+      # This is erroring, not sure why.
+      # - name: Coveralls
+      #   run: "php vendor/bin/php-coveralls"
+      #   if: ${{ success() }}
```
Comment (mglaman) [inline .github/workflows/main.yml:31]: This was failing: https://github.com/mglaman/PHP-Parser/runs/1514718327?check_suite_focus=true. Maybe due to a required secret.

**Comment by nikic** [comment]
Is there something I need to do to make it run? Doesn't look like it runs in the PR.

File: .github/workflows/main.yml (hunk 1 above)
Comment (nikic) [reply .github/workflows/main.yml:7]: It is possible to make it run for all commits/PRs, regardless of branch? Other branches don't get used much right now, as there's currently only one supported version, but generally we'd want CI for everything.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      - name: "Checkout"
+        uses: "actions/checkout@v2"
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          coverage: "none"
+          php-version: "8.0"
+          tools: composer:v2
+      - name: "Install PHP 8 dependencies"
+        run: "composer update --ignore-platform-req=php --no-progress --prefer-dist"
```
Comment (nikic) [inline .github/workflows/main.yml:69]: The ignore-platform-req is probably not needed anymore?

File: .github/workflows/main.yml (hunk 1 above)
Comment (mglaman) [reply .github/workflows/main.yml:7]: Cool; can do. This is something I've copied around – it runs on `push` to any branch but only PRs on `master`. I can make sure it runs on all things

**Comment by mglaman** [comment]
So I _thought_ it would run on the PR, but not until it's merged. So it currently only runs on the branch push for my fork. After this is merged things would start running.

**Comment by mglaman** [comment]
Re-run after suggestions addressed: https://github.com/mglaman/PHP-Parser/actions/runs/409233894

**Comment by nikic** [comment]
Looks like it does work fine once merged: https://github.com/nikic/PHP-Parser/actions/runs/409258444

The T_DOUBLE_COLON issue sounds like ext-tokenizer isn't enabled. That's probably due to the `-n` flag in https://github.com/nikic/PHP-Parser/blob/5e36ef732e0753b6e780f4d6160edf5e0869d73d/test_old/run-php-src.sh#L5. IIRC that flag is there to disable xdebug, but with the new setup it can probably just be dropped?

**Comment by nikic** [comment]
Dropped the `-n` flag in https://github.com/nikic/PHP-Parser/commit/0f64504317ea99029915203924160cbd24400bdd, build is green now.

Only remaining question is the coverage generation.

**Comment by nikic** [comment]
Possibly need to specify COVERALLS_REPO_TOKEN: https://github.com/php-coveralls/php-coveralls#github-actions

**Comment by nikic** [comment]
Coverage enabled in https://github.com/nikic/PHP-Parser/commit/7c09e096c9ec2cbfb16c072914ff25368d866002 and Travis dropped in https://github.com/nikic/PHP-Parser/commit/bec74aceda6264c79cbc7c5d2a578c8822811ab4. Looks like everything is working fine!

**Comment by mglaman** [comment]
Awesome!

**Comment by TomasVotruba** [comment]
@mglaman Thank you :+1:

**Pull request merged**